aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecorator.kt

```diff
@@ -76,25 +85,39 @@ class SigV4AuthDecorator : ClientCodegenDecorator {
         codegenContext: ClientCodegenContext,
         baseCustomizations: List<ServiceRuntimePluginCustomization>,
     ): List<ServiceRuntimePluginCustomization> =
-        baseCustomizations + listOf(AuthServiceRuntimePluginCustomization(codegenContext))
+        baseCustomizations.letIf(applies(codegenContext)) {
```

Comment:
use `extendIf` here and other places you have the "add something if pattern"

Reply:
No longer doing these conditionals after refactoring to a ConditionalDecorator.